### PR TITLE
Type fixes

### PR DESCRIPTION
**Pull request by NelsonVides**
Solves #1

### CODE REVIEW

File: src/erl_csv_decoder.erl [hunk 1]
```diff
@@ -64,33 +68,38 @@ get_more_stream(Stream) ->
...
+    {no_trailer, Match :: iodata()} |
+    {has_trailer, Match :: iodata(), Trailer :: iodata()}.
 process_chunk([], Chunk, Filtered, #csv_decoder{}, [], Acc, LenProcessed) ->
     Size = byte_size(Chunk),
     NotProcessed = Size - LenProcessed,
     NewChunk = binary:part(Chunk, Size, - NotProcessed),
-    case iolist_to_binary([NewChunk, Filtered]) of
-        <<>> -> {lists:reverse(Acc), no_trailer};
-        Trailer -> {lists:reverse(Acc), Trailer}
+    case iolist_size(NewChunk) of
```
Comment (chrzaszcz) [inline src/erl_csv_decoder.erl:100]: `Filtered` is now an unused argument

File: src/erl_csv_decoder.erl [hunk 2]
```diff
@@ -64,33 +68,38 @@ get_more_stream(Stream) ->
             NewStream
     end.
 
-% -spec process_match({error, term()} | match(), csv_decoder(), binary()) ->
-%     {ok, binary()} | {has_trailer, iolist(), binary()} | {nomatch, binary()} | {error, term()}.
+-spec process_match(
+        {match, matches()} | {error, term()} | match | nomatch, csv_decoder(), iodata()) ->
```
Comment (chrzaszcz) [inline src/erl_csv_decoder.erl:72]: The `match` atom does not make sense to me...

**Review by chrzaszcz — COMMENTED** [review]
Looks good in general, I added some comments.

File: src/erl_csv_decoder.erl (hunk 2 above)
Comment (NelsonVides) [reply src/erl_csv_decoder.erl:72]: It's on the return values of the `re:run`, dialyzer and eqwalizer forced me to add that value here before they'd stop complaining ☹️

**Review by chrzaszcz — APPROVED** [review]
Looks good :ok_hand:

**Pull request merged**